src/app/g3w-ol/controls/basequerypolygoncontrol.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @since v3.8
+ */
+import { SPATIALMETHODS, VM } from 'g3w-ol/constants';
+
+const InteractionControl = require('g3w-ol/controls/interactioncontrol');
+const { merge }          = require('core/utils/ol');
+const { Geometry }       = require('core/utils/geo');
+
+const VALIDGEOMETRIES    = Geometry.getAllPolygonGeometryTypes();
+
+const BaseQueryPolygonControl = function(options = {
+    spatialMethod=SPATIALMETHODS[0],
+    name,
+    tipLabel,
+    label,
+    onSelectlayer,
+    enabled=true,
+    interactionClass
+  } = {}) {
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.layers = options.layers || [];
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.externalLayers = [];
+
+  /**
+   * @type {unknown[]}
+   *
+   * @since 3.8.0
+   */
+  this.unwatches = [];
+
+  this.listenLayersVisibilityChange();
+
+  /**
+   * @type {boolean}
+   */
+  options.visible = this.checkVisibile(this.layers);
+
+  /**
+   * Store current selected layer
+   *
+   * @since 3.8.0
+   */
+  this.selectedLayer = null;
+
+  const _options = {
+    offline: false,
+    name,
+    tipLabel,
+    label,
+    // update selected layer
+    onSelectlayer,
+    clickmap: true, // set ClickMap
+    interactionClass,
+    spatialMethod,
+    toggledTool:{
+      type: 'spatialMethod',
+      how: 'toggled' // or hover
+    },
+    onhover: true
+  };
+
+  options = merge(options, _options);
+
+  options.geometryTypes = VALIDGEOMETRIES;
+
+  InteractionControl.call(this, options);
+
+  // starting disabled
+  this.setEnable(enabled);
+
+  this._handleExternalLayers();
```

Comment:
The `_handleExternalLayers` function it seems to be undefined within this file: `src/app/g3w-ol/controls/basequerypolygoncontrol.js`